server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/SlackMessageService.java

```diff
@@ -0,0 +1,165 @@
+package de.tum.in.www1.hephaestus.leaderboard;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import com.slack.api.bolt.App;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.chat.ChatPostMessageRequest;
+import com.slack.api.methods.response.auth.AuthTestResponse;
+import com.slack.api.methods.response.chat.ChatPostMessageResponse;
+import com.slack.api.model.User;
+import com.slack.api.model.block.LayoutBlock;
+import org.apache.commons.text.similarity.LevenshteinDistance;
+
+import static com.slack.api.model.block.Blocks.*;
+import static com.slack.api.model.block.composition.BlockCompositions.*;
+
+@Order(value = Ordered.LOWEST_PRECEDENCE)
+@EnableScheduling
+@Service
+public class SlackMessageService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackMessageService.class);
+
+    @Value("${slack.channelId}")
+    private String channelId;
+
+    @Value("${slack.runScheduledMessage}")
+    private boolean runScheduledMessage;
+
+    @Value("${spring.url:localhost:8080}")
```

Comment:
I used the `spring` section, because it contains other details regarding the "setup" as well, such as the name, Keycloak-URL, ... .
Moved it to `hephaestus.host_url` according to your suggestion in df27a34 now.